src/cdk/a11y/focus-monitor/focus-monitor.ts

```diff
@@ -36,6 +39,26 @@ export interface FocusOptions {
   preventScroll?: boolean;
 }
 
+/** Detection mode used for attributing the origin of a focus event. */
+export const enum FocusMonitorDetectionMode {
+  /**
+   * Any mousedown, keydown, or touchstart event that happened in the previous
+   * tick or the current tick will be used to assign a focus event's origin (to
+   * either mouse, keyboard, or touch). This is the default option.
+   */
+  IMMEDIATE,
+  /**
+   * A focus event's origin is always attributed to the last corresponding
+   * mousedown, keydown, or touchstart event, no matter how long ago it occured.
+   */
+  EVENTUAL
+}
+
+/** InjectionToken that can be used to specify the detection mode. */
+export const FOCUS_MONITOR_DETECTION_MODE =
```

Comment:
Could you make this a config option called `FOCUS_MONITOR_DEFAULT_OPTIONS` with `detectionMode` as the only option for now? We're trying to be more consistent with how components/services are configured via provider to reduce the number of tokens.

Reply:
Done.